lib/java/com/google/android/material/floatingactionbutton/ExtendedFloatingActionButton.java

```diff
@@ -1092,7 +1124,7 @@ public void onAnimationEnd() {
       super.onAnimationEnd();
       animState = ANIM_STATE_NONE;
       if (!isCancelled) {
-        setVisibility(GONE);
+        setVisibility(INVISIBLE);
```

Comment:
Is this required?

Reply:
Yes it is required. If you don't make it invisible then the ext fab loses its layout position. And then when it reanimates back in, it starts in the bottom right corner and quickly jumps to the center line. This happens because alpha is being animated, but technically the view is not marked as `VISIBLE` so the layout position is not calculated.

Reply:
the `performNow()` method can set it to GONE because it's not animating? Or should that also be INVISIBLE.

Reply:
I would say keep it since issues do not arise from it currently. 

Because I am contributing, I think as a general philosophy I should probably only address issue (or PR) specific issues. Architectural improvements can be saved for another issue / PR combo. Otherwise you get PR #798 😉 

Sorry, just a drawback of working with a good ol' contributor...

Reply:
Yeah I don't want to increase the scope of this PR, but I do want to make sure we aren't missing a case and this bug is fully fixed. I think `performNow()` is called instead if the view isn't laid out yet, wouldn't that mean it would lose its layout position if that happens? Or is something else happening that fixes this?